src/bit-systems/object-menu.ts

```diff
@@ -39,10 +52,69 @@ function moveToTarget(world: HubsWorld, menu: EntityID) {
   const targetObj = world.eid2obj.get(ObjectMenu.targetRef[menu])!;
   targetObj.updateMatrices();
 
+  // TODO: position the menu more carefully...
+  //       For example, if a menu object is just placed at a target
+  //       object's position the menu object can be hidden by a large
+  //       target object or the menu object looks too small for a far
+  //       target object.
+  _mat4.copy(targetObj.matrixWorld);
+
+  // Keeps world scale (1, 1, 1) because
+  // a menu object is a child of a target object
+  // and the target object's scale can be changed.
+  // Another option may be making the menu object
+  // a sibling of the target object.
+  _mat4.decompose(_vec3_1, _quat, _vec3_2);
+  _vec3_2.set(1.0, 1.0, 1.0);
+  _mat4.compose(_vec3_1, _quat, _vec3_2);
+
   const menuObj = world.eid2obj.get(menu)!;
+  setMatrixWorld(menuObj, _mat4);
 
-  // TODO: position the menu more carefully...
-  setMatrixWorld(menuObj, targetObj.matrixWorld);
+  // TODO: Remove the dependency with AFRAME
+  const camera = AFRAME.scenes[0].systems["hubs-systems"].cameraSystem.viewingCamera;
+  camera.updateMatrices();
+  menuObj.lookAt(_vec3_1.setFromMatrixPosition(camera.matrixWorld));
+}
+
+// TODO: startRotation/Scaling() and stopRotation/Scaling() are
+//       temporary implementation that rely on the old systems.
+//       They should be rewritten more elegantly with bitecs.
+function startRotation(world: HubsWorld, targetEid: EntityID) {
+  // TODO: Don't use any
+  // TODO: Remove the dependency with AFRAME
+  const transformSystem = (AFRAME as any).scenes[0].systems["transform-selected-object"];
+  const physicsSystem = AFRAME.scenes[0].systems["hubs-systems"].physicsSystem;
+  physicsSystem.updateBodyOptions(Rigidbody.bodyId[targetEid], { type: "kinematic" });
```

Comment:
Not something to address here but just to get it bouncing around in everyone's head. We change these physics properties pretty arbitrarily all over and its very easy for things to end up in weird states and to overwrite one system's desired physics settings with another. We need to think of how to manage that. It mostly comes about when something wants to set the position/rotation of a physics object since it must be kinematic or static to do so, but then it needs to be able to return to its previous state.